### PR TITLE
edit workflow to add support for ppc64le image generation

### LINKED ISSUE #1419: Make rabbitmq/cluster-operator image available for ppc64le architecture
**Is your feature request related to a problem? Please describe.**
rabbitmq/cluster-operator is not supported on ppc64le architecture. Images are built only for amd64 and arm64. 

**Describe the solution you'd like**
I would like to run the RabbitMQ operator on ppc64le machine using image from dockerhub

**Describe alternatives you've considered**
Built ppc64le image locally and used it from privet registry

**Additional context**
RabbitMQ [image](https://hub.docker.com/layers/library/rabbitmq/latest/images/sha256-578aa9026155c6d1c9f2af3773e53ec065b6026d3e1f72087247f243ef4eca58?context=explore) itself is multi-arch image and supports ppc64le. So supporting operator image as well on ppc64le will be added advantage

### PR DESCRIPTION
**Pull request by Sapana-Khemkar**
This closes #1419

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes
This PR edits github actions workflow and add support to generate ppc64le image along with amd64 and arm64. 
## Additional Context
None

## Local Testing

Please ensure you run the unit, integration and system tests before approving the PR.

To run the unit and integration tests:

```
$ make unit-tests integration-tests
```

You will need to target a k8s cluster and have the operator deployed for running the system tests.

For example, for a Kubernetes context named `dev-bunny`:
```
$ kubectx dev-bunny
$ make destroy deploy-dev
# wait for operator to be deployed
$ make system-tests
```

### CODE REVIEW

File: .github/workflows/build-test-publish.yml [hunk 1]
```diff
@@ -155,7 +155,30 @@ jobs:
...
+          type=sha,suffix=-ppc64le,latest=false
+    - name: Build and push single-arch ppc64le image
+      #if: startsWith(github.ref, 'refs/tags/v')
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        platforms: linux/ppc64le
+        push: ${{ github.event_name != 'pull_request' }}
+        tags: ${{ steps.single-arch-meta-ppc64le.outputs.tags }}
+        labels: ${{ steps.single-arch-meta-ppc64le.outputs.labels }}
```
Comment (Zerpet) [inline .github/workflows/build-test-publish.yml:181]: We don't need a special case for single-arch image for ppc64. The other single-arch image cases are to workaround a limitation from one of our internal processes. The ppc64 arch won't flow as part of that internal process, so this case won't be needed

**Review by Zerpet — CHANGES_REQUESTED** [review]
Thank you for contributing to the cluster operator! The changes are in the right direction. I left a comment to remove two steps that won't be necessary.

**Comment by Zerpet** [comment]
CI fails because the fork doesn't have access to the repo secrets. The Operator builds fine, and that should be enough. The entire pipeline will run after merging.

File: .github/workflows/build-test-publish.yml (hunk 1 above)
Comment (Sapana-Khemkar) [reply .github/workflows/build-test-publish.yml]: @Zerpet  Noted and removed single arch image case for ppc64le

**Review by Zerpet — APPROVED** [review]

**Comment by Zerpet** [comment]
Thank you!

**Comment by Sapana-Khemkar** [comment]
@Zerpet  when is your next release planned so we can avoid using sha value docker pull

**Comment by Zerpet** [comment]
There's not much to release a new minor yet

https://github.com/rabbitmq/cluster-operator/compare/v2.4.0...main

Probably after summer.

**Pull request merged**